x/gov/proposal/handler.go

```diff
@@ -0,0 +1,45 @@
+package proposal
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov/tags"
+)
+
+// HandleSubmit submits content and does initial deposit with provided SubmitForm
+func HandleSubmit(ctx sdk.Context, k Keeper,
+	content Content, proposer sdk.AccAddress, initialDeposit sdk.Coins) sdk.Result {
+	// XXX: ValidateSubmitProposalBasic is supposed to be called in each msg types
+	// It it inefficient to do it here again?
+	// I think we still should do, proposer can still skip CheckTx()
```

Comment:
We have the guarantee that `ValidateBasic` will have been called in `baseapp.runTx`

Reply:
The case I am worried about is when the implementors miss to include `ValidateSubmitProposalBasic()` in their `msg.ValidateBasic()`. It won't be an issue for now, I will remove XXX but will leave the paragraph.

Will there be any way we can enforce to include it in the messages?

Reply:
It depends on who's implementing the messages.